### ISSUE TITLE
External subsystems and optional variables break with 2DOF paramport

### ISSUE THREAD

**Issue by jkirk5** [OpenMDAO/Aviary#434]
### Description

The 2DOF ODEs use `paramport`, which has a hardcoded list of variables (`param_data`) that are used to avoid input value/unit conflicts (among other possible applications?). This breaks when the problem contains unexpected variables needed by the ODE - this is confirmed to happen when using a propeller-based aircraft, and will also probably happen with external subsystems (which don't appear to be added to 2DOF odes at all?).

The solution is unclear but it is possible a refactor is necessary to avoid this issue.

### Example

Try running `test_bench_large_turboprop.py` from here (https://github.com/jkirk5/om-Aviary/blob/regional_cargo_model/aviary/validation_cases/benchmark_tests/test_bench_turboprop_freighter.py), as of commit 
cc724b2.

Basically any 2DOF benchmark with a turboprop added instead of an EngineDeck will replicate the issue

### Aviary Version

0.9.4-dev

**Comment by jkirk5** [comment]
Due to recent changes to paramport, this is no longer breaking the provided example model
